src/js/scheduler.ts

```diff
@@ -1,9 +1,14 @@
 import { RUNTIME_ENV } from "./environments";
+
 const scheduleCallbackImmediateMessagePrefix =
   "sched$" + Math.random().toString(36).slice(2) + "$";
+
 const tasks: Record<number, () => void> = {};
 let nextTaskHandle = 0;
 
+let sharedChannel: MessageChannel | null = null;
+const queue: number[] = [];
```

Comment:
nitpick: the name `queue` sounds too general, how about something like `taskQueue` or `schedulerQueue`?